### ISSUE TITLE
Fix price overflow

### ISSUE THREAD

**Issue by nymkappa** [mempool/mempool#3214]
<img width="204" alt="image" src="https://user-images.githubusercontent.com/9780671/222868093-15238bfd-3368-4e5a-85ef-1dfc06c19b05.png">

**Comment by nymkappa** [comment]
Related https://github.com/mempool/mempool/issues/3180

I think in some widget like the "Latest transactions" widget, we could use power of ten display when amount is getting too big.

For example, if a transaction value is `1,234,567` fiat unit, we could display `1,234M` instead, something like that. We already use the same idea for difficulty adjustments, or avg tx fee for example. So that would be easy to implement.

<img width="139" alt="image" src="https://user-images.githubusercontent.com/9780671/222938873-1ba3ce88-abcb-40df-9aba-1abb9364198a.png">

**Comment by algodondev** [comment]
Good day, @nymkappa do you think it would be proper to modify the raw fiat-component so it gets the fixing on every place? I ask this because some people may want to know the entire transaction details even when it doesn't fit in the screen.

Also, was related #3180 fixed? If not I can go with both inside this issue's fix.

**Comment by nymkappa** [comment]
> Good day, @nymkappa do you think it would be proper to modify the raw fiat-component so it gets the fixing on every place? I ask this because some people may want to know the entire transaction details even when it doesn't fit in the screen.

Yes, but it needs to be configurable. Maybe take my PR as it was before it got closed and fix the issue raised by wiz (support different locale).


> Also, was related #3180 fixed? If not I can go with both inside this issue's fix.

Looks like it was already fixed yes.

**Comment by algodondev** [comment]
Understood and good to know, thank you!

**Comment by algodondev** [comment]
I'm about to start on this @nymkappa . Doing some research (Google Gemini) I think it would be better to use only K and M shorthands since the next ones may be hard to understand wolrdwide (for example, for latin-america, billions are not a thousand millions but a million millions). What do you think about it?

**Comment by nymkappa** [comment]
@algodondev does not javascript have native support for power of ten notation based on locale?
